src/nrfDevice.ts

```diff
@@ -136,31 +121,41 @@ export const nrfDevice = (
   });
 
   client.on('connect', async () => {
-    log.success('connected');
-    await notifyOfConnection('connect');
+    await handleConnect('connect');
   });
 
   client.on('message', (topic: string, payload: any) => {
     log.incoming(topic, payload || {});
     const p = payload ? JSON.parse(payload.toString()) : {};
+    if (topic.lastIndexOf('/c2d') > 0) {
+      topic = device.topics.c2d;
```

Comment:
Why this? Seems to cut off any subtopics of `/c2d`

Reply:
It's  b/c the subscription uses a wildcard, so the topic could end in anything, but the way listeners are registered requires a constant value like `c2d`. So I'm just ensuring that, regardless of the c2d topic the message is received on, it will evaluate to the key expected by the listener map.